syntax = "proto3";

package chat;

message ChatRoomRequest {
  string room_id = 1;
  string user_id = 2;
  string message = 3;
}

message ChatRoomResponse {
  string room_id = 1;
  string user_id = 2;
  string message = 3;
}

// 채팅 메시지 요청
message ChatMessageRequest {
  string room_id = 1;
  string user_id = 2;
  string message = 3;
}

// 서버의 응답 메시지
message ChatMessageResponse {
  string status = 1;
  string message = 2;
}

// gRPC 서비스 정의
service ChatService {
  rpc CreateRoom(ChatRoomRequest) returns (ChatRoomResponse);
  rpc JoinRoom(ChatRoomRequest) returns (ChatRoomResponse);
  rpc LeaveRoom(ChatRoomRequest) returns (ChatRoomResponse);
  rpc Message(stream ChatMessageRequest) returns (stream ChatMessageResponse);
}